lux/vislib/altair/BarChart.py

```diff
@@ -40,44 +40,55 @@ def initialize_chart(self):
         x_attr = self.vis.get_attr_by_channel("x")[0]
         y_attr = self.vis.get_attr_by_channel("y")[0]
 
+        x_attr_abv = x_attr.attribute
+        y_attr_abv = y_attr.attribute
+
+        if len(x_attr.attribute) > 25:
+            x_attr_abv = x_attr.attribute[:15] + "..." + x_attr.attribute[-10:]
+        if len(y_attr.attribute) > 25:
+            y_attr_abv = y_attr.attribute[:15] + "..." + y_attr.attribute[-10:]
+
+        x_attr.attribute = x_attr.attribute.replace(".", "")
```

Comment:
Why is the replace "." done once in AltairRenderer.py and once again inside each of the chart type? Is this necessary?

Reply:
this is renaming the actual object attribute parameter. i believe altair Renderer changes the column names but does not change the x_attr.attribute